lms/djangoapps/course_home_api/outline/v1/tests/test_views.py

```diff
@@ -0,0 +1,46 @@
+"""
+Tests for Outline Tab API in the Course Home API
+"""
+
+
+import ddt
+
+from django.urls import reverse
+
+from course_modes.models import CourseMode
+from lms.djangoapps.course_home_api.tests.utils import BaseCourseHomeTests
+from student.models import CourseEnrollment
+
+
+@ddt.ddt
+class OutlineTabTestViews(BaseCourseHomeTests):
+    """
+    Tests for the Outline Tab API
+    """
+    @classmethod
+    def setUpClass(cls):
+        BaseCourseHomeTests.setUpClass()
+        cls.url = reverse('course-home-outline-tab', args=[cls.course.id])
+
+    @ddt.data(CourseMode.AUDIT, CourseMode.VERIFIED)
+    def test_get_authenticated_enrolled_user(self, enrollment_mode):
+        CourseEnrollment.enroll(self.user, self.course.id, enrollment_mode)
+        response = self.client.get(self.url)
+        self.assertEqual(response.status_code, 200)
+
+        course_tools = response.data.get('course_tools')
+        if course_tools:
+            self.assertTrue(course_tools)
+            self.assertEquals(course_tools[0]['analytics_id'], 'edx.bookmarks')
+
+    def test_get_authenticated_user_not_enrolled(self):
+        response = self.client.get(self.url)
+        self.assertEqual(response.status_code, 200)
+        self.assertFalse(response.data.get('learner_is_verified'))
+
+    def test_get_unauthenticated_user(self):
+        self.client.logout()
+        response = self.client.get(self.url)
+        self.assertEqual(response.status_code, 403)
+
+    # TODO: write test_get_unknown_course when more data is pulled into the Outline Tab API
```

Comment:
This is a TODO because test_get_unknown_course was returning a 200 rather than a 404. I think this is because when there are no Course Tools available for a course, it returns an empty list. I didn't think an empty list was necessarily a blocker to loading a course, and that this test should just be written when the API is fleshed out a bit more. Thoughts?

Reply:
But if you're trying to get a course that doesn't even exist, it doesn't 404 on that get? That feels odd to me.

Reply:
Ah, you never request the course as part of the API since you only need the key for now. Very well.